elements/lisk-transactions/src/15_proof_of_misbehavior_transaction.ts

```diff
@@ -0,0 +1,445 @@
+			minimum: 0,
+		},
+		totalAmount: {
+			type: 'string',
+			format: 'amount',
+		},
+		totalFee: {
+			type: 'string',
+			format: 'amount',
+		},
+		reward: {
+			type: 'string',
+			format: 'amount',
+		},
+		seedReveal: {
+			type: 'string',
+			format: 'hex',
+		},
+		payloadHash: {
+			type: 'string',
+			format: 'hex',
+		},
+		timestamp: {
+			type: 'integer',
+			minimum: 0,
+		},
+		numberOfTransactions: {
+			type: 'integer',
+			minimum: 0,
+		},
+		payloadLength: {
+			type: 'integer',
+			minimum: 0,
+		},
+		previousBlockId: {
+			type: ['null', 'string'],
+			format: 'id',
+			minLength: 1,
+			maxLength: 20,
+		},
+		generatorPublicKey: {
+			type: 'string',
+			format: 'publicKey',
+		},
+		maxHeightPrevoted: {
+			type: 'integer',
+			minimum: 0,
+		},
+		maxHeightPreviouslyForged: {
+			type: 'integer',
+			minimum: 0,
+		},
+		height: {
+			type: 'integer',
+			minimum: 1,
+		},
+		blockSignature: {
+			type: 'string',
+			format: 'signature',
+		},
+	},
+};
+
+const proofOfMisbehaviorAssetFormatSchema = {
+	type: 'object',
+	required: ['header1', 'header2'],
+	properties: {
+		header1: blockHeaderSchema,
+		header2: blockHeaderSchema,
+	},
+};
+export interface ProofOfMisbehaviorAsset {
+	readonly header1: BlockHeader;
+	readonly header2: BlockHeader;
+	// tslint:disable-next-line readonly-keyword
+	reward?: bigint;
+}
+
+export class ProofOfMisbehaviorTransaction extends BaseTransaction {
+	public readonly asset: ProofOfMisbehaviorAsset;
+	public static TYPE = 15;
+
+	public constructor(rawTransaction: unknown) {
+		super(rawTransaction);
+		const tx = (typeof rawTransaction === 'object' && rawTransaction !== null
+			? rawTransaction
+			: {}) as Partial<TransactionJSON>;
+		this.asset = (tx.asset || {}) as ProofOfMisbehaviorAsset;
+		this.asset.reward = BigInt(0);
+	}
+
+	public assetToJSON(): ProofOfMisbehaviorAsset {
+		return {
+			header1: this.asset.header1,
+			header2: this.asset.header2,
+		};
+	}
+
+	protected assetToBytes(): Buffer {
+		return Buffer.concat([
+			getBlockBytesWithSignature(this.asset.header1),
+			getBlockBytesWithSignature(this.asset.header1),
+		]);
+	}
+
+	public async prepare(store: StateStorePrepare): Promise<void> {
+		const delegateId = getAddressFromPublicKey(
+			this.asset.header1.generatorPublicKey,
+		);
+
+		const filterArray = [
+			{
+				address: this.senderId,
+			},
+			{
+				address: delegateId,
+			},
+		];
+
+		await store.account.cache(filterArray);
+	}
+
+	protected validateAsset(): ReadonlyArray<TransactionError> {
+		const asset = this.assetToJSON();
+		const schemaErrors = validator.validate(
+			proofOfMisbehaviorAssetFormatSchema,
+			asset,
+		);
+		const errors = convertToAssetError(
+			this.id,
+			schemaErrors,
+		) as TransactionError[];
+
+		if (
+			this.asset.header1.generatorPublicKey !==
+			this.asset.header2.generatorPublicKey
+		) {
+			errors.push(
+				new TransactionError(
+					'GeneratorPublickey of each blockheader must be matching.',
```

Comment:
"GeneratorPublickey of each blockheader should be the same." or "GeneratorPublickey of each blockheader should match."